File: libJava/com/earndero/sqlwrapper/java/WritableDatabase.java
package com.earndero.sqlwrapper.java;

import java.io.Closeable;
import java.io.IOException;
import java.sql.SQLException;

public class WritableDatabase extends Database implements Closeable {
    public WritableDatabase(Context context, String name) throws Exception {
        super(name);
    }

    @Override
    public void close() throws IOException {

    }

    public Table createTable(String name) {
        return new Table(false, name, this);
    }

    boolean TransactionSuccessful = false;

    public void beginTransaction() throws SQLException {
        TransactionSuccessful = false;
        connection.setAutoCommit(false);
    }

    public void setTransactionSuccessful() {
        TransactionSuccessful = true;
    }

    public void endTransaction() throws SQLException {
        if (TransactionSuccessful)
            connection.commit();
        else
            connection.rollback();
        connection.setAutoCommit(true);
    }

    public void drop(String name, boolean ifExists) throws SQLException {
        String stmt = "DROP TABLE ";
        if (ifExists) stmt+="IF EXISTS ";
        stmt+=name;
        java.sql.Statement statement  = connection.createStatement();
        statement.execute(stmt);
    }
}

File: libJavaKt/com/earndero/sqlwrapper/javakt/WritableDatabase.kt
package com.earndero.sqlwrapper.javakt

import kotlin.Throws
import java.io.IOException
import java.io.Closeable
import java.sql.SQLException

class WritableDatabase(context: Context?, name: String) : Database(name), Closeable {
    @Throws(IOException::class)
    override fun close() {
    }

    fun createTable(name: String): Table {
        return Table(false, name, this)
    }

    var TransactionSuccessful = false
    @Throws(SQLException::class)
    fun beginTransaction() {
        TransactionSuccessful = false
        connection!!.autoCommit = false
    }

    fun setTransactionSuccessful() {
        TransactionSuccessful = true
    }

    @Throws(SQLException::class)
    fun endTransaction() {
        if (TransactionSuccessful) connection!!.commit() else connection!!.rollback()
        connection!!.autoCommit = true
    }

    @Throws(SQLException::class)
    fun drop(name: String?, ifExists: Boolean) {
        var stmt: String? = "DROP TABLE "
        if (ifExists) stmt += "IF EXISTS "
        stmt += name
        val statement = connection!!.createStatement()
        statement.execute(stmt)
    }
}